index.js

```diff
@@ -174,49 +173,34 @@ class RNShare {
     return new Promise((resolve, reject) => {
       requireAndAskPermissions(options)
         .then(() => {
-          if (Platform.OS === 'ios' && !options.urls) {
-            // Handle for single file share
-            ActionSheetIOS.showShareActionSheetWithOptions(
-              options,
-              error => {
-                return reject({ error: error });
-              },
-              (success, activityType) => {
-                if (success) {
-                  return resolve({
-                    app: activityType,
-                  });
-                } else if (options.failOnCancel === false) {
-                  return resolve({
-                    dismissedAction: true,
-                  });
-                } else {
-                  reject(new Error('User did not share'));
-                }
-              },
-            );
-          } else {
-            NativeModules.RNShare.open(
-              options,
-              e => {
-                return reject({ error: e });
-              },
-              (success, activityType) => {
-                if (success) {
-                  return resolve({
-                    app: activityType,
-                    message: activityType,
-                  });
-                } else if (options.failOnCancel === false) {
-                  return resolve({
-                    dismissedAction: true,
-                  });
-                } else {
-                  reject(new Error('User did not share'));
-                }
-              },
-            );
+          if(options.url && !options.urls) // Backward compatibility with { Share } from react-native
+          {
+            const url = options.url;
+            options.url = undefined;
```

Comment:
Applying this is broking the CI. What do you think about applying a ```delete options.url```?

Reply:
I was just on it :)